13.keitai_message/keitai_message.rb

```diff
@@ -0,0 +1,44 @@
+def keitai_message(input)
+  testcase = input.split(/0+/)
+  testcase.delete('')
+
+  patterns =
+  { 1 => ['\'', ',', '!', '?', ' '],
+    2 => %w[a b c],
+    3 => %w[d e f],
+    4 => %w[g h i],
+    5 => %w[j k l],
+    6 => %w[m n o],
+    7 => %w[p q r s],
+    8 => %w[t u v],
+    9 => %w[w x y z]
+  }
+
+  result = ''
+  testcase.each do |s|
+    i = s[0].to_i
+    pattern = patterns[i]
+    case i
+    when 1
+      if s.count('1-9') % 5 == 0
```

Comment:
ここって事実上

```suggestion
      if s.count('1') % 5 == 0
```

しかありえないですよね？

ロジック上、存在しないパターンを指定されると読み手が混乱します。

Reply:
おっしゃる通りですね💧
改めてロジックを整理したところ、ここは文字数を数えられれば十分だと判断したので、`length`を使用することにしました。